File: 0x00-pascal_triangle/0-pascal_triangle.py
#!/usr/bin/python3
"""
Module with the pascals triangle generator
"""

def pascal_triangle(n):
    """generate a list of lists with pascal triangle"""
    triangle = []
    if n > 0:
        triangle = [[1]]
        n = n - 1
        for i in range(n):
            currVal = 0
            next = []
            length = len(triangle[-1])
            current = triangle[-1]
            for i in range(length):
                next.append(currVal + current[i])
                currVal = current[i]
            next.append(currVal + 0)
            triangle.append(next)
    return triangle

File: 0x00-pascal_triangle/0-pascal_triangle.js
export default function pascal_triangle(n) {
    const triangle = [];
    if (n > 0){
        triangle.push([1]);
        n = n - 1;
        for (let i = 0; i < n; i++) {
            const next = [];
            let currVal = 0;
            const currPascal = triangle[i];
            let length = currPascal.length
            for (let j = 0; j < length; j++) {
                next.push(currVal + currPascal[j]);
                currVal = currPascal[j];
            }
            next.push(currVal + 0);
            triangle.push(next);
        }
    }
    return triangle;
}
